### PR TITLE
Chore/decouple hashi executor

### PR DESCRIPTION
**Pull request by mani99brar**
## PR-Codex overview
This PR focuses on enhancing the `relayer-cli` by updating state management, improving message execution handling, and refining the code structure for better maintainability and clarity.

### Detailed summary
- Added timestamp and nonce fields to various state JSON files.
- Updated `HashiMessageExecutionVars` to include timestamp.
- Introduced new bot events for message failing and indexing.
- Refactored state file functions to handle pending messages.
- Improved message execution logic to handle THRESHOLD_NOT_MET status.
- Simplified state file updates by removing hashiBlockNumber.
- Enhanced tests to cover new functionalities and edge cases.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Summary by CodeRabbit

* **New Features**
  * Added Arbitrum One → Story bridge route
  * Persistent relayer state and pending-message queuing across runs

* **Improvements**
  * More robust batch execution: per-message checks, failing messages excluded and reported
  * Improved indexing with capped ranges and per-message timestamps
  * New events for indexing progress and message failure; improved local logging visibility

### CODE REVIEW

File: relayer-cli/src/utils/hashi.ts [hunk 1]
```diff
@@ -41,39 +47,69 @@ interface HashiExecutorInterface {
...
+      continue;
+    }
+    if (messageState.executable) {
+      executableMessages.push(messageState.hashiMessage);
+    } else if (
+      messageState.status === HashiExecutionStatus.THRESHOLD_NOT_MET &&
+      tx.timestamp + MAX_PENDING_TIME_SECONDS >= Math.floor(Date.now() / 1000)
+    ) {
+      pendingMessages.push(tx);
+    }
```
[the remark this answers is not in the document]
Comment (mani99brar) [reply relayer-cli/src/utils/hashi.ts:83]: Updated the toExecuteMessage to return the full msgStatus

File: relayer-cli/src/utils/hashi.ts [hunk 2]
```diff
@@ -99,50 +135,65 @@ async function executeBatchOnHashi(
...
+        await yaru.executeMessages.staticCall([message]);
+        filteredMessages.push(message);
+      } catch {
+        emitter.emit(
+          BotEvents.HASHI_MESSAGE_FAILING,
+          message.nonce.toString(),
+          sourceChainId.toString(),
+          targetChainId.toString()
+        );
+      }
```
[the remark this answers is not in the document]
Comment (mani99brar) [reply relayer-cli/src/utils/hashi.ts:188]: We shouldn't add it to pending if the msg execution is failing even after the threshold is met.

File: relayer-cli/src/utils/hashi.ts [hunk 3]
```diff
@@ -221,18 +272,24 @@ async function getAllMessageDispatchedLogs(
...
+  emitter.emit(BotEvents.INDEXING, fromBlock, toBlock);
   const iface = new Interface(messageDispatchedAbi);
   const topic0 = iface.getEvent("MessageDispatched").topicHash;
 
   const all: HashiMessageExecutionVars[] = [];
 
   let start = fromBlock;
-  while (start <= toBlock) {
+  while (start <= toBlock && all.length <= MAX_BATCH_SIZE) {
     const end = Math.min(start + chunkSize - 1, toBlock);
```
[the remark this answers is not in the document]
Comment (mani99brar) [reply relayer-cli/src/utils/hashi.ts:295]: 👍

File: relayer-cli/src/utils/hashiHelpers/stateFile.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  setLock(network, chainId);
+  emitter.emit(BotEvents.LOCK_CLAIMED);
+  // STATE_DIR is absolute path of the directory where the state files are stored
+  // STATE_DIR must have trailing slash
+  const stateDir = process.env.STATE_DIR || "";
+  const stateFile = path.join(stateDir, `${network}_${chainId}.json`);
+  if (!fileSystem.existsSync(stateFile)) {
+    // No state file so initialize starting now
+    const tsnow = Math.floor(Date.now() / 1000);
+    await syncStateFile(chainId, tsnow, 0, [], network, emitter);
```
[the remark this answers is not in the document]
Comment (mani99brar) [reply relayer-cli/src/utils/hashiHelpers/stateFile.ts]: Done

File: relayer-cli/src/utils/hashiHelpers/stateFile.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  if (!fileSystem.existsSync(stateDir)) {
+    fileSystem.mkdirSync(stateDir, { recursive: true });
+  }
+  const chain_state_file = process.env.STATE_DIR + network + "_" + chainId + ".json";
+  const json = {
+    ts: createdTimestamp,
+    hashiBlockNumber: hashiBlockNumberFrom,
+    hashiMessages: hashiMessages,
+  };
+  fileSystem.writeFileSync(chain_state_file, JSON.stringify(json, null, 2), { encoding: "utf8" });
```
[the remark this answers is not in the document]
Comment (mani99brar) [reply relayer-cli/src/utils/hashiHelpers/stateFile.ts]: Done 👍

File: relayer-cli/state/testnet_10200.json [hunk 6]
```diff
@@ -1,5 +1,4 @@
 {
-  "ts": 1768221100,
-  "nonce": 0,
-  "hashiBlockNumber": 232819590
+  "ts": 1777270251,
+  "nonce": 12
```
[the remark this answers is not in the document]
Comment (mani99brar) [reply relayer-cli/state/testnet_10200.json:3]: It's done deliberately to sync the bot with blockchain state and there's no message to execute before the nonce.

File: relayer-cli/src/utils/hashiHelpers/stateFile.ts [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  const json = {
+    ts: createdTimestamp,
+    hashiBlockNumber: hashiBlockNumberFrom,
+    hashiMessages: hashiMessages,
+  };
+  fileSystem.writeFileSync(chain_state_file, JSON.stringify(json, bigIntReplacer, 2), { encoding: "utf8" });
+  if (!isIniting) {
+    removeLock(network, chainId);
+  }
+  emitter.emit(BotEvents.LOCK_RELEASED);
```
[the remark this answers is not in the document]
Comment (mani99brar) [reply relayer-cli/src/utils/hashiHelpers/stateFile.ts:84]: Noted.

**Review by jaybuidl — APPROVED** [review]
